src/zenml/integrations/mlflow/model_deployers/mlflow_model_deployer.py

```diff
@@ -0,0 +1,307 @@
+        service_directory_path = existing_service.status.runtime_path
+        shutil.rmtree(service_directory_path)
+
+
+    def _existing_service_exists(
+        self,
+        existing_service: MLFlowDeploymentService, 
+        config: MLFlowDeploymentConfig
+    ) -> bool:
+        """Returns true if an old service with the same pipeline name,
+        step name and model name exists.
+        
+        Args:
+            existing_service: The materialized Service instance derived from the config
+            of the older (existing) service
+            config: The MLFlowDeploymentConfig object passed to the deploy_model function holding 
+            parameters of the new service to be created."""
+        
+        existing_service_config = existing_service.config
+
+        if existing_service_config.pipeline_name == config.pipeline_name and \
+            existing_service_config.model_name == config.model_name and \
+            existing_service_config.step_name == config.step_name:
+            return True
+
+        return False
+    
+    # the step will receive a config from the user that mentions the number of workers etc.
+    # the step implementation will create a new config using all values from the user and 
+    # add values like pipeline name, model_uri 
+    def _create_new_service(
+        self, 
+        timeout: int,
+        config: MLFlowDeploymentConfig
+    ) -> MLFlowDeploymentService:
+        """Creates a new MLFlowDeploymentService."""
+
+        # set the uuid in the config
+        config.caller_uuid = self.uuid
+        # create a new service for the new model
+        service = MLFlowDeploymentService(config)
+        service.start(timeout=timeout)
+
+        return service
+
+        
+    def find_model_server(
+        self,
+        pipeline_name: Optional[str] = None,
+        run_id: Optional[str] = None,
+        step_name: Optional[str] = None,
+        model_name: Optional[str] = None,
+        model_uri: Optional[str] = None,
+        model_type: Optional[str] = None,
+    ) -> List[MLFlowDeploymentService]:
+        """Method to find one or more model servers that match the
+        given criteria.
+
+        Args:
+            pipeline_name: Name of the pipeline that the deployed model was part
+            of.
+            run_id: ID of the pipeline run which the deployed model was part of.
+            step_name: The name of the pipeline model deployment step that
+                deployed the model.
+            model_name: Name of the deployed model.
+            model_uri: URI of the deployed model.
+            model_type: Type/format of the deployed model. Not used in this MLflow
+                case.
+
+        Returns:
+            One or more Service objects representing model servers that match
+            the input search criteria.
+        """
+
+        services = []
+        config = MLFlowDeploymentConfig(
+            model_name=model_name,
+            model_uri=model_uri,
```

Comment:
nice catch! Thank you

Reply:
What if I modify the mlflow deployment config to have `model_uri: str = ""` instead of `model_uri: str`. Will that work? I think that will make the code in the model deployer above look cleaner and consistent with the rest of the parameters.